### ISSUE TITLE
Automatic cleartext blocklist refresh

### ISSUE THREAD

**Issue by obtuse** [globaleaks/Tor2web#282]
It might be useful to have automatic blocklist updates for the cleartext blocklist like we do for the hashed version. This might make it easier for many people, especially those working in teams, to update the blocklist.

**Comment by evilaliv3** [comment]
i see your point @obtuse; you mean fetching it from a remote source right?

i would find critical starting sharing that publicly links to blocked contents.
you do not think so?

**Comment by obtuse** [comment]
Making the cleartext blocklist public is definitely a bad idea. But we can easily host a cleartext blocklist somewhere privately or even on localhost.

Is it possible to provide an easier way to update the blocklist, or block new URLs?

**Comment by evilaliv3** [comment]
I find it risky but that would be eventually a one line change doing at line https://github.com/globaleaks/Tor2web/blob/master/tor2web/t2w.py#L96 the same done at line https://github.com/globaleaks/Tor2web/blob/master/tor2web/t2w.py#L89

@virgil @fpietrosanti what do you think?

**Comment by obtuse** [comment]
I actually tried the single line change on OnionLink but it does not work because we need the `block_list` variable to be updated, which requires a daemon reload (since `load_lists()` needs to be called in order to import the contents of `blocklist_cleartext` into `block_list`). So it might require a little more work to update `block_list` after `cleartext_blocklist` is updated.

Or maybe we can use the config.automatic_blocklist_updates_refresh setting to implement a looping call on `load_lists` itself, which would allow all blocklists to be updated each time?

**Comment by evilaliv3** [comment]
right; we should eventually implement this chaining effect.

let's wait whatt the others would say anyhow of such a feature.

are you already planning to have multiple servers using this?

**Comment by evilaliv3** [comment]
and what blocks you to simply use distribute hashed lists?

**Comment by obtuse** [comment]
I agree, we need to find out if there are others who would benefit from this feature.

Hashed lists work fine for us as far as functionality is concerned. However if two or more people manage the blocklist, maintaining a cleartext blocklist is much more convenient.
